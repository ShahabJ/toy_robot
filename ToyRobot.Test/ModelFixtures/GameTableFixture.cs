using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using ToyRobot.Library.Model;
using ToyRobot.Library.Service;
using Xunit;

namespace ToyRobot.Test.ModelFixtures
{
    [UseReporter(typeof(DiffReporter))]
    public class GameTableFixture
    {
        int minX = 0;
        int minY = 0;
        int maxX = 6;
        int maxY = 6;

        [Fact]
        public void ExecureReport()
        {
            var table = new GameTable(minX, minY, maxX, maxY);
            var result = new List<string>();
            //Header for a test report
            result.Add($"Assertion for GameTable: Min X:{minX},Min Y:{minY}. Max X:{maxX}, Y:{maxY}");

            for (int i = 0; i < 10; i++)
            {
                for (int j = 0; j < 10; j++)
                {
                    var onTheTable = table.IsOnTheTable(new Position(i, j, DirectionEnum.NORTH)) ? "OnTheTable" : "NotOnTheTable";
                    result.Add($"X:{i},y:{j} => {onTheTable}");
                }
            }

            Approvals.VerifyAll(result, "");
        }
    }
}