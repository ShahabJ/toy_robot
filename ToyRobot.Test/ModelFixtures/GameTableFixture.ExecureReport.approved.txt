[0] = Assertion for GameTable: Min X:0,Min Y:0. Max X:6, Y:6
[1] = X:0,y:0 => OnTheTable
[2] = X:0,y:1 => OnTheTable
[3] = X:0,y:2 => OnTheTable
[4] = X:0,y:3 => OnTheTable
[5] = X:0,y:4 => OnTheTable
[6] = X:0,y:5 => OnTheTable
[7] = X:0,y:6 => NotOnTheTable
[8] = X:0,y:7 => NotOnTheTable
[9] = X:0,y:8 => NotOnTheTable
[10] = X:0,y:9 => NotOnTheTable
[11] = X:1,y:0 => OnTheTable
[12] = X:1,y:1 => OnTheTable
[13] = X:1,y:2 => OnTheTable
[14] = X:1,y:3 => OnTheTable
[15] = X:1,y:4 => OnTheTable
[16] = X:1,y:5 => OnTheTable
[17] = X:1,y:6 => NotOnTheTable
[18] = X:1,y:7 => NotOnTheTable
[19] = X:1,y:8 => NotOnTheTable
[20] = X:1,y:9 => NotOnTheTable
[21] = X:2,y:0 => OnTheTable
[22] = X:2,y:1 => OnTheTable
[23] = X:2,y:2 => OnTheTable
[24] = X:2,y:3 => OnTheTable
[25] = X:2,y:4 => OnTheTable
[26] = X:2,y:5 => OnTheTable
[27] = X:2,y:6 => NotOnTheTable
[28] = X:2,y:7 => NotOnTheTable
[29] = X:2,y:8 => NotOnTheTable
[30] = X:2,y:9 => NotOnTheTable
[31] = X:3,y:0 => OnTheTable
[32] = X:3,y:1 => OnTheTable
[33] = X:3,y:2 => OnTheTable
[34] = X:3,y:3 => OnTheTable
[35] = X:3,y:4 => OnTheTable
[36] = X:3,y:5 => OnTheTable
[37] = X:3,y:6 => NotOnTheTable
[38] = X:3,y:7 => NotOnTheTable
[39] = X:3,y:8 => NotOnTheTable
[40] = X:3,y:9 => NotOnTheTable
[41] = X:4,y:0 => OnTheTable
[42] = X:4,y:1 => OnTheTable
[43] = X:4,y:2 => OnTheTable
[44] = X:4,y:3 => OnTheTable
[45] = X:4,y:4 => OnTheTable
[46] = X:4,y:5 => OnTheTable
[47] = X:4,y:6 => NotOnTheTable
[48] = X:4,y:7 => NotOnTheTable
[49] = X:4,y:8 => NotOnTheTable
[50] = X:4,y:9 => NotOnTheTable
[51] = X:5,y:0 => OnTheTable
[52] = X:5,y:1 => OnTheTable
[53] = X:5,y:2 => OnTheTable
[54] = X:5,y:3 => OnTheTable
[55] = X:5,y:4 => OnTheTable
[56] = X:5,y:5 => OnTheTable
[57] = X:5,y:6 => NotOnTheTable
[58] = X:5,y:7 => NotOnTheTable
[59] = X:5,y:8 => NotOnTheTable
[60] = X:5,y:9 => NotOnTheTable
[61] = X:6,y:0 => NotOnTheTable
[62] = X:6,y:1 => NotOnTheTable
[63] = X:6,y:2 => NotOnTheTable
[64] = X:6,y:3 => NotOnTheTable
[65] = X:6,y:4 => NotOnTheTable
[66] = X:6,y:5 => NotOnTheTable
[67] = X:6,y:6 => NotOnTheTable
[68] = X:6,y:7 => NotOnTheTable
[69] = X:6,y:8 => NotOnTheTable
[70] = X:6,y:9 => NotOnTheTable
[71] = X:7,y:0 => NotOnTheTable
[72] = X:7,y:1 => NotOnTheTable
[73] = X:7,y:2 => NotOnTheTable
[74] = X:7,y:3 => NotOnTheTable
[75] = X:7,y:4 => NotOnTheTable
[76] = X:7,y:5 => NotOnTheTable
[77] = X:7,y:6 => NotOnTheTable
[78] = X:7,y:7 => NotOnTheTable
[79] = X:7,y:8 => NotOnTheTable
[80] = X:7,y:9 => NotOnTheTable
[81] = X:8,y:0 => NotOnTheTable
[82] = X:8,y:1 => NotOnTheTable
[83] = X:8,y:2 => NotOnTheTable
[84] = X:8,y:3 => NotOnTheTable
[85] = X:8,y:4 => NotOnTheTable
[86] = X:8,y:5 => NotOnTheTable
[87] = X:8,y:6 => NotOnTheTable
[88] = X:8,y:7 => NotOnTheTable
[89] = X:8,y:8 => NotOnTheTable
[90] = X:8,y:9 => NotOnTheTable
[91] = X:9,y:0 => NotOnTheTable
[92] = X:9,y:1 => NotOnTheTable
[93] = X:9,y:2 => NotOnTheTable
[94] = X:9,y:3 => NotOnTheTable
[95] = X:9,y:4 => NotOnTheTable
[96] = X:9,y:5 => NotOnTheTable
[97] = X:9,y:6 => NotOnTheTable
[98] = X:9,y:7 => NotOnTheTable
[99] = X:9,y:8 => NotOnTheTable
[100] = X:9,y:9 => NotOnTheTable
